### PR TITLE
chore(git hook): add precommit-hook to handle validation checks

### LINKED ISSUE #880: [FEAT] Add git hooks with husky
**Is your feature request related to a problem? Please describe.**

- The feature is unrelated to a problem but will help automate the development cycle. We can use Husky to lint commit messages, run tests, and lint code when you commit or push code. We can specify commit message formats, add pre-commit and post-merge hooks

**Describe the solution you'd like**

- There are specific tasks to handle with git hooks for this project: 
  1. Automate code linting after staging file changes
  2.  Running automated tests

**Additional context**

### PR DESCRIPTION
**Pull request by EspiraMarvin**
- This change adds validation check for linting, formatting & test concurrently on commit

Fixes #880

### CODE REVIEW

**Comment by ronzulu** [comment]
Hi @EspiraMarvin does this mean that the lint/test operations will occur whenever a developer pushes local commits to GitHub. I.e. these operations will occur regardless of whether or not there is an associated PR?

Thanks
Ronny

File: package-lock.json
Comment (st3v3nmw) [inline package-lock.json:1]: Looks like you used `npm` instead of `pnpm`

**Review by st3v3nmw — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (EspiraMarvin) [reply package-lock.json:1]: @st3v3nmw I'll change it to use pnpm

**Comment by ronzulu** [comment]
> @ronzulu Yes, according to the PR, the operations will run regardless of whether or not there is an associated PR.

Thanks. I think this is really helpful.

**Comment by st3v3nmw** [comment]
@EspiraMarvin 👀 ...

**Comment by EspiraMarvin** [comment]
> @EspiraMarvin 👀 ...

@st3v3nmw  this is ready

**Review by st3v3nmw — APPROVED** [review]
LGTM, thanks!

**Pull request merged**